### PR TITLE
Update ECDSA signature conversion specification

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Update the spec following https://github.com/Mbed-TLS/mbedtls/pull/8703#discussion_r1474697136 resolved by 315e4afc0a6bc4e55340fe8de7891e076e277da5.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** no (doc only)
- [x] **backport** no (new stuff)
- [x] **tests** N/A (doc only)

### CODE REVIEW

**Review by mpg — APPROVED** [review]

File: docs/architecture/psa-migration/psa-legacy-bridges.md [hunk 1]
```diff
@@ -330,12 +330,12 @@ Based on the [gap analysis](#signature-formats):
...
-                             size_t bits);
+int mbedtls_ecdsa_raw_to_der(size_t bits,
+                             const unsigned char *raw, size_t raw_len,
+                             unsigned char *der, size_t der_size, size_t *der_len);
+int mbedtls_ecdsa_der_to_raw(size_t bits,
+                             const unsigned char *der, size_t der_len,
+                             unsigned char *raw, size_t raw_size, size_t *raw_len);
 ```
 
 * These functions convert between the signature format used by `mbedtls_pk_{sign,verify}{,_ext}` and the signature format used by `psa_{sign,verify}_{hash,message}`.
```
Comment (valeriosetti) [inline docs/architecture/psa-migration/psa-legacy-bridges.md:341]: [This is a comment for the 3rd bullet point below; I add it here because I cannot do this in the proper place.]
Even though this belongs to a previous PR, I would rephrase

>The `bits` parameter is not needed in the raw-to-DER direction, but [it can help catch errors](https://github.com/Mbed-TLS/mbedtls/pull/8681#discussion_r1445980971) and the information is readily available in practice.

as:

> The `bits` parameter might look redundant in the raw-to-DER direction, but [it helps catching errors](https://github.com/Mbed-TLS/mbedtls/pull/8681#discussion_r1445980971) so it's kept anyway.

Wdyt?

**Review by valeriosetti — COMMENTED** [review]
First of all thanks for adding this change. I totally forgot to do it in #8703.
I left only 1 minor comment for a possible rephrasing. If you think that it is not relevant or it's wrong, just let me know and I will approve the PR as-is ;)

**Review by valeriosetti — APPROVED** [review]
LGTM ;)

**Review by mpg — APPROVED** [review]
LGTM!
